tools/cli/go-whisk-cli/commands/action.go

```diff
@@ -404,41 +404,23 @@ func parseAction(cmd *cobra.Command, args []string, update bool) (*whisk.Action,
             }
         }
 
-        if flags.action.kind == "swift:3" || flags.action.kind == "swift:3.0" || flags.action.kind == "swift:3.0.0" {
-            action.Exec.Kind = "swift:3"
-        } else if flags.action.kind == "nodejs:6" || flags.action.kind == "nodejs:6.0" ||
-                flags.action.kind == "nodejs:6.0.0" {
-            action.Exec.Kind = "nodejs:6"
-        } else if flags.action.kind == "nodejs:default" {
-            action.Exec.Kind = "nodejs:default"
-        } else if flags.action.kind == "swift:default" {
-            action.Exec.Kind = "swift:default"
-        } else if flags.action.kind == "nodejs" {
-            action.Exec.Kind = "nodejs"
-        } else if flags.action.kind == "python" {
-            action.Exec.Kind = "python"
+        if len(flags.action.kind) > 0 {
```

Comment:
Yes it will:

```
> wsk action create foo t.js --kind foobar
error: Unable to create action: The request content was malformed:
kind 'foobar' not in Set(swift:3, nodejs, blackbox, java, sequence, nodejs:6, python, swift) (code 1206)
Run 'wsk --help' for usage.
```

I don't understand the second part of the comment re parseAction.